ggml.h

```diff
@@ -457,6 +479,7 @@ extern "C" {
 
     GGML_API size_t  ggml_set_scratch (struct ggml_context * ctx, struct ggml_scratch scratch);
     GGML_API void    ggml_set_no_alloc(struct ggml_context * ctx, bool no_alloc);
+    GGML_API void    ggml_set_default_backend(struct ggml_context * ctx, enum ggml_backend backend);
```

Comment:
We should avoid adding this function / state - it's too much GPU logic for `ggml`.

The `ggml_tensor->backend` + `ggml_tensor->extra` should be everything that we need for GPU backends.
Otherwise we start complicating the core `ggml` logic with GPU state which I really want to avoid.

Also, the `backend` and `extra` members should not be referenced inside `ggml.c`.
They are available to the user code and the backend implementations (`ggml-cuda`, `ggml-opencl`, etc.).
This is a restriction that we should follow in order to avoid the GPU logic to proliferate in the core library.